### PR TITLE
[Docs] Add documentation for username_template feature in AWS Secrets Engine

### PR DESCRIPTION
**Pull request by vinay-gopalan**
Adds documentation about the `username_template` feature, including length limits, errors and an example conditional template.

### CODE REVIEW

File: website/content/api-docs/secret/aws.mdx [hunk 1]
```diff
@@ -58,6 +58,16 @@ valid AWS credentials with proper permissions.
...
 
+- `username_template` `(string: <optional>)` - [Template](/docs/concepts/username-templating) describing how
+  dynamic usernames are generated. The username template is used to generate both IAM usernames (capped at 64 characters) 
+  and STS usernames (capped at 32 characters). Longer usernames result in a 500 error.
+
+  To ensure generated usernames are within length limits for both STS/IAM, the template must adequately handle
+  both conditional cases (see [Conditional Templates](https://pkg.go.dev/text/template)). As an example, if no template
+  is provided the field defaults to the template:
+
+  `{{ if (eq .Type "STS") }}{{ printf "vault-%s-%s"  (unix_time) (random 20) | truncate 32 }}{{ else }}{{ printf "vault-%s-%s-%s" (printf "%s-%s" (.DisplayName) (.PolicyName) | truncate 42) (unix_time) (random 20) | truncate 64 }}{{ end }}`
```
Comment (jasonodonnell) [inline website/content/api-docs/secret/aws.mdx:69]: Trying to make this easier to read:
```suggestion
```
{{ if (eq .Type "STS") }}
  {{ printf "vault-%s-%s" (unix_time) (random 20) | truncate 32 }}
{{ else }}
  {{ printf "vault-%s-%s-%s" (printf "%s-%s" (.DisplayName) (.PolicyName) | truncate 42) (unix_time) (random 20) | truncate 64 }}{{ end }}
```
```

Comment (jasonodonnell) [reply website/content/api-docs/secret/aws.mdx:69]: I guess it won't work in a suggestion:
```
{{ if (eq .Type "STS") }}
  {{ printf "vault-%s-%s" (unix_time) (random 20) | truncate 32 }}
{{ else }}
  {{ printf "vault-%s-%s-%s" (printf "%s-%s" (.DisplayName) (.PolicyName) | truncate 42) (unix_time) (random 20) | truncate 64 }}
{{ end }}
```

Comment (jasonodonnell) [inline website/content/api-docs/secret/aws.mdx:69]: For readability can we display this as:
{{ if (eq .Type "STS") }}
    {{ printf "vault-%s-%s" (unix_time) (random 20) | truncate 32 }}
{{ else }}
    {{ printf "vault-%s-%s-%s" (printf "%s-%s" (.DisplayName) (.PolicyName) | truncate 42) (unix_time) (random 20) | truncate 64 }}
{{ end }}

**Review by jasonodonnell — APPROVED** [review]
LGTM, just a small suggestion for the default template doc.

**Review by pcman312 — APPROVED** [review]
LGTM - I agree with @jasonodonnell 's suggestion too

File: website/content/api-docs/secret/aws.mdx (hunk 1 above)
Comment (vinay-gopalan) [reply website/content/api-docs/secret/aws.mdx]: Fixed!

File: website/content/api-docs/secret/aws.mdx [hunk 2]
```diff
@@ -58,6 +58,22 @@ valid AWS credentials with proper permissions.
...
+  both conditional cases (see [Conditional Templates](https://pkg.go.dev/text/template)). As an example, if no template
+  is provided the field defaults to the template:
+
+  `
+  {{ if (eq .Type "STS") }}
+      {{ printf "vault-%s-%s" (unix_time) (random 20) | truncate 32 }}
+  {{ else }}
+      {{ printf "vault-%s-%s-%s" (printf "%s-%s" (.DisplayName) (.PolicyName) | truncate 42) (unix_time) (random 20) | truncate 64 }}
+  {{ end }}
+  `
```
Comment (jasonodonnell) [inline website/content/api-docs/secret/aws.mdx:75]: These need to be triple ticks, its not rendering right without them: https://vault-3zc820va4-hashicorp.vercel.app/api-docs/secret/aws#username_template

Comment (vinay-gopalan) [reply website/content/api-docs/secret/aws.mdx]: Yup, just realized. Fixed!

**Pull request merged**